pkg/cri/server/images/image_pull.go

```diff
@@ -334,10 +334,27 @@ func (c *CRIImageService) createImageReference(ctx context.Context, name string,
 	} else if !errdefs.IsAlreadyExists(err) {
 		return err
 	}
-	if oldImg.Target.Digest == img.Target.Digest && oldImg.Labels[crilabels.ImageLabelKey] == labels[crilabels.ImageLabelKey] {
-		return nil
+	// Retrieve oldImg from image store here because Create routine returns an
+	// empty image on ErrAlreadyExists
+	oldImg, err := c.images.Get(ctx, name)
+	if err != nil {
+		return err
+	}
+	if oldImg.Target.Digest == img.Target.Digest {
+		var updated bool
+		if oldImg.Labels[crilabels.ImageLabelKey] != labels[crilabels.ImageLabelKey] {
+			_, err = c.images.Update(ctx, img, "target", "labels."+crilabels.ImageLabelKey)
+			updated = true
```

Comment:
works but sort of odd to pass this err down out of the statement block to the check below on 358.. vs just returning an error here.. 

is there a possibility to have both imagelabelkey and pinnedimagelabel key not equivalent to getlabels()?  if so an err on this update would be overwritten by an error on the pinned update .. and it does the update twice...  

looks like Update() supports a list of fieldpaths () if we need to update both labels .. suggest calling update just once so we only have to write once..